src/garrysmod/gamemodes/zombiesurvival/gamemode/cl_scoreboard.lua

```diff
@@ -1,216 +1,249 @@
+        surface.SetDrawColor(5, 5, 5, 180)
+        surface.DrawRect(0, 0, w, h)
+        surface.SetDrawColor(90, 90, 90, 180)
+        self:DrawOutlinedRect()
+
+        local barw = 64
+        surface.SetDrawColor(5, 5, 5, 160)
+        surface.DrawRect(w * 0.5 - 16, 0, 32, h - 64)
+        surface.SetTexture(texRightEdge)
+        surface.DrawTexturedRect(w * 0.5 + 16, 0, barw, h - 64)
+        surface.DrawTexturedRectRotated(w * 0.5 - 16 - barw / 2, (h - 64) / 2, barw, h - 64, 180)
+        surface.SetTexture(texCorner)
+        surface.DrawTexturedRectRotated(w * 0.5 - 16 - barw / 2, h - 32, barw, 64, 90)
+        surface.DrawTexturedRectRotated(w * 0.5 + 16 + barw / 2, h - 32, barw, 64, 180)
+        surface.SetTexture(texDownEdge)
+        surface.DrawTexturedRect(w * 0.5 - 16, h - 64, 32, 64)
+    end
+
+    self.humanContainer = vgui.Create("Panel")
+    self.grid:AddCell(self.humanContainer)
+
+    self.m_HumanHeading = vgui.Create("DTeamHeading", self.humanContainer)
+    self.m_HumanHeading:SetTeam(TEAM_HUMAN)
+    self.m_HumanHeading:Dock(TOP)
+    self.m_HumanHeading:DockMargin(0, 0, 8, 0)
+
+    self.humanColumnNames = vgui.Create("ZSPlayerPanelHeader", self.humanContainer)
+    self.humanColumnNames:Dock(TOP)
+    self.humanColumnNames:DockMargin(0, 0, 8, 0)
+
+    self.HumanList = vgui.Create("DScrollPanel", self.humanContainer)
+    self.HumanList:Dock(FILL)
+    self.HumanList:DockMargin(0, 0, 8, 0)
+
+    self.zombieContainer = vgui.Create("Panel")
+    self.grid:AddCell(self.zombieContainer)
+
+    self.m_ZombieHeading = vgui.Create("DTeamHeading", self.zombieContainer)
+    self.m_ZombieHeading:SetTeam(TEAM_UNDEAD)
+    self.m_ZombieHeading:Dock(TOP)
+    self.m_ZombieHeading:DockMargin(8, 0, 0, 0)
+
+    self.zombieColumnNames = vgui.Create("ZSPlayerPanelHeader", self.zombieContainer)
+    self.zombieColumnNames:Dock(TOP)
+    self.zombieColumnNames:DockMargin(8, 0, 0, 0)
+    self.zombieColumnNames:SetTeam(TEAM_UNDEAD)
+
+    self.ZombieList = vgui.Create("DScrollPanel", self.zombieContainer)
+    self.ZombieList:Dock(FILL)
+    self.ZombieList:DockMargin(8, 0, 0, 0)
+
+    self:InvalidateLayout()
 end
 
-function PANEL:PerformLayout()
-	local screenscale = math.max(0.95, BetterScreenScale())
-
-	self.m_AuthorLabel:MoveBelow(self.m_TitleLabel)
-	self.m_ContactLabel:MoveBelow(self.m_AuthorLabel)
-
-	self.m_ServerNameLabel:SetPos(math.min(self:GetWide() - self.m_ServerNameLabel:GetWide(), self:GetWide() * 0.75 - self.m_ServerNameLabel:GetWide() * 0.5), 32 - self.m_ServerNameLabel:GetTall() / 2)
-
-	self.m_HumanHeading:SetSize(self:GetWide() / 2 - 32, 28 * screenscale)
-	self.m_HumanHeading:SetPos(self:GetWide() * 0.25 - self.m_HumanHeading:GetWide() * 0.5, 110 * screenscale - self.m_HumanHeading:GetTall())
+function PANEL:PerformLayout(w, h)
+    local screenscale = math.max(0.95, BetterScreenScale())
+    self.m_Header:SetSize(w, 32)
+    self.grid:SetRowHeight(h - 32 - self.grid:GetVerticalMargin() * 1.5)
```

Comment:
These are the only static values in the entire scoreboard. Everything else fills in.